deepinv/models/epll.py

```diff
@@ -0,0 +1,457 @@
+import torch.nn as nn
+import torch
+from deepinv.utils import patch_extractor
+from deepinv.optim.utils import conjugate_gradient
+from .utils import get_weights_url
+from deepinv.physics import Denoising, GaussianNoise
+from tqdm import tqdm
+
+
+class EPLL(nn.Module):
+    r"""
+    Patch prior via Gaussian mixture models.
+
+    The forward method evaluates the negative log likelihood of the GMM.
+    The reconstruction function implements the approximated half-quadratic splitting method as in Zoran, D., and Weiss, Y.  "From learning models of natural image patches to whole image restoration." (ICCV 2011).
+
+    :param deepinv.models.GaussianMixtureModel GMM: Gaussian mixture defining the distribution on the patch space.
+        None creates a GMM with n_components components of dimension accordingly to the arguments patch_size and channels.
+    :param int n_components: number of components of the generated GMM if GMM is None.
+    :param str pretrained: Path to pretrained weights of the GMM with file ending .pt. None for no pretrained weights, "download" for pretrained weights on the BSDS500 dataset, "GMM_lodopab_small" for the weights from the limited-angle CT example.
```

Comment:
If you can add a pointer to `deepinv.models.rst` that specifies the refs for the pretrained models, that would be amazing!